### PR TITLE
WIP: Gym/Battle

### PR DESCRIPTION
**Pull request by Grover-c13**
WIP: Battle + Gym API

### CODE REVIEW

**Comment by Grover-c13** [comment]
Just incase someone else decides to start work on it while im not on my PC :P.

Currently some issues with it, i am guessing to do it properly you need to start a battle, then join a battle. Currently getting a bad result returned though so really need some network dumps of a full gym battle.

File: src/main/java/com/pokegoapi/api/gym/Battle.java [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+		for (int i = 0; i < team.length; i++) {
+			builder.addAttackingPokemonIds(team[i].getId());
+		}
+
+
+		List<PokemonDataOuterClass.PokemonData> defenders = gym.getDefendingPokemon();
+		builder.setGymId(gym.getId());
+		builder.setPlayerLongitude(api.getLongitude());
+		builder.setPlayerLatitude(api.getLatitude());
+		builder.setDefendingPokemonId(defenders.get(0).getId()); // may need to be sorted
```
Comment (vmarchaud) [inline src/main/java/com/pokegoapi/api/gym/Battle.java:92]: L#78 : they are already from what i saw, but you may check if the gym is actually defended (no team has claim it), its could lead to a NPE

File: src/main/java/com/pokegoapi/api/gym/Battle.java [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+		}
+
+	}
+
+	public Result start() throws LoginFailedException, RemoteServerException {
+
+		Builder builder = StartGymBattleMessageOuterClass.StartGymBattleMessage.newBuilder();
+
+		for (int i = 0; i < team.length; i++) {
+			builder.addAttackingPokemonIds(team[i].getId());
```
Comment (vmarchaud) [inline src/main/java/com/pokegoapi/api/gym/Battle.java:84]: L#70 : You could just use addAllAttackingPokemonIds() methods

**Comment by vmarchaud** [comment]
Network dump of a gym attack: http://pastebin.com/enDzFvUN

**Pull request merged**